### PR TITLE
Added Fail2ban for Dokuwiki

### PR DESCRIPTION
**Pull request by anmol26s**
## Problem
- *App don't have Fail2ban*

## Solution
- *Added Fail2ban*

## PR Status
- [X] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [X] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [ ] **Code review** : 
- [ ] **Approval (LGTM)** : 
- [ ] **Approval (LGTM)** :  
- **CI succeeded** : [Build Status](https://ci-apps-hq.yunohost.org/jenkins/job/dokuwiki_ynh%20PR44/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by anmol26s** [comment]
This needs to tested. Works for me. 
And maybe regular expression could be made more sophisticated.

**Comment by maniackcrudelis** [comment]
Looks good, but need to be tested.

**Comment by frju365** [comment]
We need to set up a test on the dev CI.

File: scripts/change_url [hunk 1]
```diff
@@ -75,14 +75,17 @@ then
 	ynh_add_nginx_config
 fi
 
-# Change the domain for nginx
+# Change the domain for nginx and impliment Fail2ban
```
Comment (Gofannon) [inline scripts/change_url]: Typo here
impliment => implement

**Comment by anmol26s** [comment]
Works for me on root.

**Comment by maniackcrudelis** [comment]
> Works for me on root.

Are you sure this feature is working ?
I've just tested it, dokuwiki isn't logging anything in nginx log !
And actually, there's a plugin made for that purpose, but it's currently broken.

Plugin [logautherror](https://www.dokuwiki.org/plugin:logautherror), and the current [PR](https://github.com/Astrel/dokuwiki_plugin_logautherror/pull/2) which is suppose to fix it.

**Comment by anmol26s** [comment]
@maniackcrudelis 
You might had checked access log for the domain. The error is reported in nginx-error log of the domain.
https://github.com/YunoHost-Apps/dokuwiki_ynh/pull/44/commits/70d8aea655dc578370b16f21ecd41bd7b4d31cd1#diff-ad962c6fd65b725b71aa46c8a715127fR218

**Comment by maniackcrudelis** [comment]
Definitively not, there's no errors log both in access or error log of the domain.
I can try many times to crack the password, more than 5 for sure. And finally with the right password I logged in.
So, fail2ban isn't working.
I think you should have a look to this plugin.

**Comment by anmol26s** [comment]
@maniackcrudelis 
You are right. My instance have plug-in which debugs authenticate log in the error log.
On fresh install it fails. 
How is Ldap plug-in installed while install. Just coping the plug-in files in the /var/www/dokuwiki/lib/plugin will do?

**Comment by maniackcrudelis** [comment]
Looks like you can simply add a plugin into the directory.
https://www.dokuwiki.org/plugin_installation_instructions#manual_instructions

**Comment by maniackcrudelis** [comment]
Fixed and tested.
Got banned after 5 attempts.

Can be reviewed

File: scripts/change_url [hunk 2]
```diff
@@ -75,14 +75,17 @@ then
...
 if [ $change_domain -eq 1 ]
 then
 	# Delete file checksum for the old conf file location
 	ynh_delete_file_checksum "$nginx_conf_path"
 	mv $nginx_conf_path /etc/nginx/conf.d/$new_domain.d/$app.conf
 	# Store file checksum for the new config file location
 	ynh_store_file_checksum "/etc/nginx/conf.d/$new_domain.d/$app.conf"
+
+	# Update Fail2ban configuration
+	ynh_add_fail2ban_config --logpath="/var/log/nginx/$new_domain-error.log" --failregex="^.*authentication failure. while reading response header from upstream, client: <HOST>,.*POST $path_url.*$" --max_retry=5
```
Comment (kay0u) [inline scripts/change_url]: I think we have to update the configuration when only path_url is modified. So perhaps we should update the conf outside this `if`?

Comment (maniackcrudelis) [reply scripts/change_url]: Done

**Comment by kay0u** [comment]
Can be merged in 3 days

**Pull request merged**